golos.publication/golos.publication.cpp

```diff
@@ -745,6 +762,7 @@ fixp_t publication::calc_available_rshares(name voter, int16_t weight, uint64_t
 
 void publication::set_vote(name voter, const structures::mssgid& message_id, int16_t weight) {
     require_auth(voter);
+    send_closemssgs_trx();
```

Comment:
I think it's better to close some posts in place.
If we use deferred transaction for this, then:
- current voter change sum of rshares in reward pool
- posts closed (and this logic give a reward for post a bit less than expected)

In `on_transfer` this logic give a bit larger.

Reply:
@afalaleev what do you think?

Reply:
I agree.
Method should call close_messages(), which should reschedule itself if not-closed posts still exist